src/main/webapp/components/workspaces/workspaces.js

```diff
@@ -59,6 +68,181 @@ const workspaceById = gql`
     }
   }
 `
+const queryAttributes = gql`
+  fragment QueryAttributes on MetacardAttributes {
+    title
+    cql
+    metacard_tags
+    metacard_type
+  }
+`
+
+const useCreateQuery = workspaceId => {
+  const [queryId, setQueryId] = React.useState('')
```

Comment:
useCreateQuery shouldn't need to know about the queryId

Reply:
Not sure I follow.  The query ID is needed when the workspace is updated on line 105.  It's also used to execute the useEffect in the case that the user hit `Search` after creating the query itself.

Reply:
useCreateQuery having access to the queryId feels like a code smell to me. I think the logic of associating a query to a workspace should be handled by the graphql side